balancer/leastrequest/leastrequest.go

```diff
@@ -155,15 +155,18 @@ type picker struct {
 
 func (p *picker) Pick(balancer.PickInfo) (balancer.PickResult, error) {
 	var pickedSC *scWithRPCCount
+	var pickedSCNumRPCs int32
 	for i := 0; i < int(p.choiceCount); i++ {
 		index := grpcranduint32() % uint32(len(p.subConns))
 		sc := p.subConns[index]
 		if pickedSC == nil {
 			pickedSC = &sc
+			pickedSCNumRPCs = pickedSC.numRPCs.Load()
 			continue
 		}
-		if sc.numRPCs.Load() < pickedSC.numRPCs.Load() {
+		if sc.numRPCs.Load() < pickedSCNumRPCs {
```

Comment:
Should `sc.numRPCS.Load()` be cached too? to save one more `Load`

Reply:
Oh good point. Switched.